Feature/main architecture

JaimeMathias opened the pull request:
Implemented a Entity for the Environment Variables, made a Service to get them from Environment, initialized and logged the variables on console at Main(). Also done a unit tests that verifies if the variable from Environment is coming correctly and another to test when that variable is null.

Unit testing results:
+--------+--------+--------+--------+
| Module | Line   | Branch | Method |
+--------+--------+--------+--------+
| App    | 56,55% | 70,45% | 90%    |
+--------+--------+--------+--------+

JaimeMathias committed:
getting all environment variables

JaimeMathias committed:
separates variables in classes

JaimeMathias committed:
show variables in console

JaimeMathias committed:
fix(git-ignore): remove bin/Debug

JaimeMathias committed:
fix(): httpVerb console log

JaimeMathias committed:
implements xUnit tests

JaimeMathias committed:
implements NullException when deserialize

JaimeMathias committed:
refactores all the project

JaimeMathias committed:
changes project's folder structure

JaimeMathias committed:
refactores and makes new unit tests

JaimeMathias committed:
adds one more unit test for language

silva-willian commented:
Colocar na restrição do git as pastas .vscode/,  **/bin e **/obj e o arquivo .env

silva-willian commented on App/Program.cs:
```diff
@@ -0,0 +1,54 @@
+            System.Console.WriteLine("RoutePattern: " + data.RoutePattern);
+            System.Console.WriteLine("Versioned: " + data.Versioned);
+            System.Console.Write("HttpVerbs: ");
+            foreach (var value in data.HttpVerbs) {
```
Não usar Console.WriteLine, o .net core tem uma biblioteca especifica para log, utilizar ela

silva-willian commented:
Coloquem na restrição do git o arquivo de coverage dos testes

silva-willian commented:
Cade o readme do projeto ?

raphael-fleury committed:
Gitignore updated

raphael-fleury committed:
Removing .env from git

LuizeraDev committed:
Adding Logger

Making loggers to show messages and improving the code.

xUnit test new results:
+--------+--------+--------+--------+
| Module | Line | Branch | Method |
+--------+--------+--------+--------+
| App | 59,58% | 77,27% | 90% |
+--------+--------+--------+--------+

LuizeraDev committed:
Fixing file name and gitignore

LuizeraDev committed:
Fixing exceptions, code review

LuizeraDev committed:
Throwing Messages Exceptions by Enum

Changed string message by Enum

LuizeraDev committed:
Fixing and Improving Code for Tests
